esp32-wroom-rp/src/protocol.rs

```diff
@@ -95,7 +102,92 @@ pub(crate) struct NinaLargeArrayParam {
     data: Vec<u8, MAX_NINA_PARAM_LENGTH>,
 }
 
-impl NinaParam for NinaByteParam {
+pub(crate) struct NinaAbstractParam {
+    // Byte representation of length of data
+    length_as_bytes: [u8; 2],
```

Comment:
@calebbourg can we get rid of this field now and create the `[u8; 2]` on the fly in a getter method?

Reply:
It isn't always the case where we actually send a 2 byte representation of the param length. In most cases it's a single byte. So it made sense to have the more specific `NinaParam` types know how to determine their own `length_as_bytes` and then attach it here as part of the `into` conversion. 
I think it's worth exploring whether or not the NINA firmware could understand _always_ sending a 2 byte representation of param length. If that is possible then I think we could remove this and just calculate the `length_as_bytes` on the fly as you suggested by taking the `u16` representation of `length` and converting it to 2 bytes.